students/yixingxu/lesson06/test_mailroom4.py

```diff
@@ -0,0 +1,46 @@
+#!/usr/bin/env python
+import mailroom4
+# from cigar_party import cigar_party2 as cigar_party
+# from cigar_party import cigar_party3 as cigar_party
+                 
+def test_list_name():
+    donation_history = [
+                    {'name':'Anna','donations':[100,200,300]},
+                    {'name':'Bob','donations':[1000,2000]},
+                    {'name':'Chuck','donations':[10000]},
+                    {'name':'David','donations':[1]},
+                    {'name':'Ethan','donations':[10,20]}
+                 ]
+    expected = ['Anna', 'Bob', 'Chuck', 'David', 'Ethan']
+    assert (expected == mailroom4.list_name()) is True
+    
+def test_create_a_report():
+    donation_history = [
+                    {'name':'Anna','donations':[100,200,300]},
+                    {'name':'Bob','donations':[1000,2000]},
+                    {'name':'Chuck','donations':[10000]},
+                    {'name':'David','donations':[1]},
+                    {'name':'Ethan','donations':[10,20]}
+                 ]
+    expected = [['Chuck', '$', 10000, 1, '$', 10000.0],
+                ['Bob', '$', 3000, 2, '$', 1500.0],
+                ['Anna', '$', 600, 3, '$', 200.0],
+                ['Ethan', '$', 30, 2, '$', 15.0],
+                ['David', '$', 1, 1, '$', 1.0]]
+    assert (expected == mailroom4.create_a_report()) is True
+    
+def test_create_summary():
+    donation_history = [
+                    {'name':'Anna','donations':[100,200,300]},
+                    {'name':'Bob','donations':[1000,2000]},
+                    {'name':'Chuck','donations':[10000]},
+                    {'name':'David','donations':[1]},
+                    {'name':'Ethan','donations':[10,20]}  
+                    ]
```

Comment:
this isn't used.

Reply:
function create_summary() has a default input, which i why I put the default input donation_history in the test function. create_summary() is from line 99 to line 106 in mailroom4.py

Reply:
![screen shot 2018-05-22 at 3 31 35 pm](https://user-images.githubusercontent.com/1928658/40393677-4e5f2dc6-5dd5-11e8-9dec-f250b146f648.png)